### PR TITLE
Update explainer path for Ambassador

### LINKED ISSUE #1548: Istio creates different path to Ambassador for explainer
When launching a loanclassifier model, the path created in the virtualservice is: `prefix: /seldon/default/loanclassifier/loanclassifier/explainer/`, wheres in ambassador it would be `prefix: /seldon/default/loanclassifier-explainer/`. Running on master (1.0.3-SNAPSHOT)

File to replicate:

```yaml
apiVersion: machinelearning.seldon.io/v1
kind: SeldonDeployment
metadata:
  labels:
    app: seldon
  name: loanclassifier
spec:
  name: loanclassifier
  predictors:
  - componentSpecs:
    - spec:
        containers:
        - image: loanclassifier:0.1
          name: model
    graph:
      children: []
      name: model
      type: MODEL
      endpoint:
        type: REST
    explainer:
      type: "AnchorTabular"
      modelUri: gs://seldon-models/sklearn/income/explainer
    name: loanclassifier
    replicas: 1
```

### PR DESCRIPTION
**Pull request by cliveseldon**
Fixes #1548 

 * Update operator Ambassador code
 * Update python wrapper client
 * Update explainer notebook

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **cliveseldon**

If they are not already assigned, you can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Mar 23 16:30:43 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1585/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1585 --build=1

**Comment by seldondev** [comment]
Mon Mar 23 16:30:48 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1585/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1585 --build=2

File: operator/controllers/ambassador.go [hunk 1]
```diff
@@ -78,9 +78,9 @@ func getAmbassadorRestConfig(mlDep *machinelearningv1.SeldonDeployment,
 	}
 
 	name := p.Name
-	if nameOverride != "" {
-		name = nameOverride
-		serviceNameExternal = nameOverride
+	if isExplainer {
+		name = name + constants.ExplainerNameSuffix
+		serviceNameExternal = serviceNameExternal + constants.ExplainerPathSuffix
```
Comment (axsaucedo) [inline operator/controllers/ambassador.go]: May be better to join with `path.Join(..)` ?

Comment (axsaucedo) [reply operator/controllers/ambassador.go]: Quite minor as the above is still with dash anyways

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1585#pullrequestreview-380515348" title="Approved">axsaucedo</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by axsaucedo** [comment]
@cliveseldon there seems to be some merge conflicts

**Comment by seldondev** [comment]
New changes are detected. LGTM label has been removed.

**Comment by seldondev** [comment]
Wed Mar 25 10:10:35 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1585/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1585 --build=4

**Comment by seldondev** [comment]
Wed Mar 25 10:12:01 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1585/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1585 --build=3

**Comment by seldondev** [comment]
Wed Mar 25 17:50:30 UTC 2020
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1585/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1585 --build=7

**Comment by seldondev** [comment]
Wed Mar 25 17:50:41 UTC 2020
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1585/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1585 --build=8

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 267f7983bfbf1d7dec9f0e28f3f200b59eff4bcd | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 267f7983bfbf1d7dec9f0e28f3f200b59eff4bcd | [link]() | `/test this`
lint | 267f7983bfbf1d7dec9f0e28f3f200b59eff4bcd | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | 267f7983bfbf1d7dec9f0e28f3f200b59eff4bcd | [link]() | `/test lint`
pr-build | 267f7983bfbf1d7dec9f0e28f3f200b59eff4bcd | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 267f7983bfbf1d7dec9f0e28f3f200b59eff4bcd | [link]() | `/test this`
lint | 267f7983bfbf1d7dec9f0e28f3f200b59eff4bcd | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Thu Mar 26 10:43:09 UTC 2020
The logs for [pr-build] [13] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1585/13.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1585 --build=13

**Comment by seldondev** [comment]
Thu Mar 26 10:43:22 UTC 2020
The logs for [lint] [14] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1585/14.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1585 --build=14

**Pull request merged**